src/partisan_peer_service_server.erl

```diff
@@ -37,85 +39,58 @@
          terminate/2,
          code_change/3]).
 
--record(state, {socket, transport}).
+-record(state, {socket, ref}).
 
-%%%===================================================================
-%%% ranch_protocol callbacks
-%%%===================================================================
+-type state_t() :: #state{}.
 
-%% @private
-start_link(ListenerPid, Socket, Transport, Options) ->
-    proc_lib:start_link(?MODULE,
-                        init,
-                        [ListenerPid, Socket, Transport, Options]).
+acceptor_init(_SockName, LSocket, []) ->
+    % monitor listen socket to gracefully close when it closes
+    MRef = monitor(port, LSocket),
+    {ok, MRef}.
 
-%% @private
-init(ListenerPid, Socket, Transport, _Options) ->
-    %% Acknowledge process initialization.
-    ok = proc_lib:init_ack({ok, self()}),
+acceptor_continue(_PeerName, Socket, MRef) ->
+    send_message(Socket, {hello, node()}),
+    gen_server:enter_loop(?MODULE, [], #state{socket=Socket, ref=MRef}).
 
-    %% Acknowledge the connection.
-    ok = ranch:accept_ack(ListenerPid),
+acceptor_terminate(Reason, _) ->
+    % Something went wrong. Either the acceptor_pool is terminating or the
+    % accept failed.
+    exit(Reason).
 
-    %% Link to the socket.
-    link(Socket),
+%% gen_server api
 
-    %% Set the socket modes.
-    ok = inet:setopts(Socket, [binary, {packet, 4}, {active, true}, {keepalive, true}]),
+init(_) ->
+    {stop, acceptor}.
```

Comment:
It is never called. It has to be exported in order to not have the behavior complain but because it uses `enter_loop` it doesn't go through the init function.